modules/core/src/main/scala/org/tessellation/infrastructure/aci/StateChannelRouter.scala

```diff
@@ -37,16 +42,25 @@ object StateChannelRouter {
     def routeInput(input: StateChannelInput): OptionT[F, Ω] =
       for {
         context <- stateChannelContexts.get(input.address).toOptionT[F]
-        result <- OptionT.liftF(runCell(input, context))
+        result <- OptionT.liftF(runCell(context, input))
       } yield result
 
-    private def runCell(input: StateChannelInput, context: StateChannelContext[F]): F[Ω] =
+    private def runCell(context: StateChannelContext[F], input: StateChannelInput) =
       for {
         cell <- context.createCell(input.bytes, hypergraphContext)
         _ <- logger.debug(s"Running cell for a state channel with address ${context.address}")
-        outputOrError <- cell.run()
-        output <- outputOrError.liftTo[F]
-      } yield output
+        cellOutputOrError <- cell.run()
+        cellOutput <- cellOutputOrError.liftTo[F]
+        interpretedOutput <- context.interpretOutput(cellOutput)
+        _ <- publishOutput(interpretedOutput)
+      } yield cellOutput
+
+    private def publishOutput(output: StateChannelOutput) =
+      output match {
+        case o @ Snapshot(_, _) => stateChannelOutputQueue.offer(o)
+        case _                  => Applicative[F].unit
```

Comment:
I guess it suggests that it can be any other output, whereas for now `StateChannelOutput` is a `sealed trait` and we know that it will be the second case `Other`. Do we intend to unseal that trait?

Reply:
Maybe we should have the special case called `StateChannelSnapshotOutput` as I guess it reflects better the restriction that we make here?

Reply:
I refactored it and there's no longer `Other` type.